### ISSUE TITLE
[Utility] Utility Module Foundation - Part 1 / 2

### ISSUE THREAD

**Issue by Olshansk** [pokt-network/pocket#475]
## Objective

Partial refactor, cleanup, and code improvement of the utility module.

## Origin Document

Originally intended as follow-up work for #325.

When work on #443 started, while also looking at #325, it became evident that the logic in the utility module is hard to follow and understand, while also being filled with minor hard-to-find bugs.

## Goals

- **Unblock development**: Create a foundation on which we design utility protocols and implement new features
- **Reduce cognitive overhead**: Improve code readability and navigation for both internal and external contributors
- **Enable onboarding other engineers**: Clarify both source code and tests through comment and source code improvements

## Deliverable

- [ ] A PR that updates a significant portion (1/N) of the utility module
- [ ] Documentation & explanation of the changes that were made and next steps

## Non-goals / Non-deliverables

- Updating the entire module in a single PR (PR would be too large)
- Adding new features to the utility module
- Increasing/improving the test coverage to the po

## General issue deliverables

- [ ] Update the appropriate CHANGELOG(s)
- [ ] Update any relevant local/global README(s)
- [ ] Update relevant source code tree explanations
- [ ] Add or update any relevant or supporting [mermaid](https://mermaid-js.github.io/mermaid/) diagrams

## Testing Methodology

- [ ] **All tests**: `make test_all`
- [ ] **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @Olshansk
